src/transport/SecureMessageCodec.cpp

```diff
@@ -43,22 +43,27 @@ CHIP_ERROR Encrypt(const CryptoContext & context, CryptoContext::ConstNonceView
     VerifyOrReturnError(!msgBuf->HasChainedBuffer(), CHIP_ERROR_INVALID_MESSAGE_LENGTH);
     VerifyOrReturnError(msgBuf->TotalLength() <= kMaxAppMessageLen, CHIP_ERROR_MESSAGE_TOO_LONG);
 
-    static_assert(std::is_same<decltype(msgBuf->TotalLength()), uint16_t>::value,
+    static_assert(std::is_same<decltype(msgBuf->TotalLength()), size_t>::value,
                   "Addition to generate payloadLength might overflow");
 
     ReturnErrorOnFailure(payloadHeader.EncodeBeforeData(msgBuf));
 
-    uint8_t * data    = msgBuf->Start();
-    uint16_t totalLen = msgBuf->TotalLength();
+    uint8_t * data  = msgBuf->Start();
+    size_t totalLen = msgBuf->TotalLength();
 
     MessageAuthenticationCode mac;
     ReturnErrorOnFailure(context.Encrypt(data, totalLen, data, nonce, packetHeader, mac));
 
     uint16_t taglen = 0;
     ReturnErrorOnFailure(mac.Encode(packetHeader, &data[totalLen], msgBuf->AvailableDataLength(), &taglen));
 
-    VerifyOrReturnError(CanCastTo<uint16_t>(totalLen + taglen), CHIP_ERROR_INTERNAL);
-    msgBuf->SetDataLength(static_cast<uint16_t>(totalLen + taglen));
+    VerifyOrReturnError(CanCastTo<size_t>(totalLen + taglen), CHIP_ERROR_INTERNAL);
```

Comment:
This check is just checking for `totalLen + taglen >= 0`, effectively, right?  But won't that always be the case?  It's an unsigned value.

The overflow checks here are broken and need to be fixed.  I brought this up in https://github.com/project-chip/connectedhomeip/pull/31776#discussion_r1509682211 too, and unlike the other comments on that PR it doesn't look like that was addressed.

Reply:
This check should have been removed because I now have bound checks for totalLen + taglen below. There were a few bits here and there that I needed to change and was doing another pass today. Had pushed the PR to check/fix builds on all platforms for CI.